src/components/announcementsList/index.tsx

```diff
@@ -1,65 +1,163 @@
+import {
+  NO_ANNOUNCEMENTS_HEADER,
+  NO_ANNOUNCEMENTS_SUBHEADER,
+} from '../../utils/copy';
 import { AnnouncementCard } from '../AnnouncementCard';
 
-const COLUMNS_PER_ROW = 3;
-const NO_ANNOUNCEMENTS_HEADER = 'There are currently no announcements!';
-const NO_ANNOUNCEMENTS_SUBHEADER = 'Come back later for future updates!';
-const NO_ANNOUNCEMENTS_HEADER_COLOR = ORANGE;
-
-const AnnouncementRow = styled(Row)`
-  display: grid;
-  grid-template-columns: 1fr 1fr 1fr;
-  margin-bottom: 20px;
-`;
 const NoAnnouncementsSubText = styled.span`
   display: block;
   text-align: center;
 `;
 
 const NoAnnouncementsText = styled(NoAnnouncementsSubText)`
-  color: ${NO_ANNOUNCEMENTS_HEADER_COLOR};
+  color: ${ORANGE};
   font-size: 36px;
   font-weight: 800;
 `;
 
+const AnnouncementsListWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  justify-content: center;
+  margin-bottom: 40px;
+`;
+
+const PageNumbersWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+
+const PageNumber = styled.li`
+  font-size: 30px;
+  list-style-type: none;
+  cursor: pointer;
+  padding: 0px 16px;
+  border: 1px solid #d9d9d9;
+  box-sizing: border-box;
+  border-radius: 4px;
+  margin: 0px 8px;
+`;
+
+const SelectedPageNumber = styled(PageNumber)`
+  color: ${LINK};
+  border: 1px solid ${LINK};
+`;
+
+const ArrowButton = styled(PageNumber)`
+  color: #bfbfbf;
+`;
+
+const noAnnouncementsTexts: JSX.Element = (
+  <>
+    <NoAnnouncementsText>{NO_ANNOUNCEMENTS_HEADER}</NoAnnouncementsText>
+    <NoAnnouncementsSubText>
+      {NO_ANNOUNCEMENTS_SUBHEADER}
+    </NoAnnouncementsSubText>
+  </>
+);
+
 export interface AnnouncementsListProps {
   announcements: Announcement[];
 }
 
 const AnnouncementsList: React.FC<AnnouncementsListProps> = ({
   announcements,
 }) => {
-  const [numRows, setNumRows] = useState(0);
+  const announcementsPerPage = 6;
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const handleNextPage = () => {
+    setCurrentPage(currentPage + 1);
+  };
+
+  const handlePreviousPage = () => {
+    setCurrentPage(currentPage - 1);
+  };
+
+  const handleNoOnClick = () => {
+    setCurrentPage(currentPage);
+  };
 
-  useEffect(() => {
-    const rows = Math.ceil(announcements.length / COLUMNS_PER_ROW);
-    setNumRows(rows);
-  }, [announcements]);
+  const handlePageClick = (pageNum: number) => {
+    setCurrentPage(pageNum);
+  };
+
+  const noAnnouncements = announcements.length === 0;
+  const indexOfLastAnnouncement = currentPage * announcementsPerPage;
+  const indexOfFirstAnnouncement =
+    indexOfLastAnnouncement - announcementsPerPage;
+  const currentAnnouncements = announcements.slice(
+    indexOfFirstAnnouncement,
+    indexOfLastAnnouncement,
+  );
+
+  const pageNumbers = [];
+  for (
+    let i = 1;
+    i <= Math.ceil(announcements.length / announcementsPerPage);
+    i++
+  ) {
+    pageNumbers.push(i);
+  }
```

Comment:
This should be updated to use array methods, I believe I left a comment elsewhere. Was the content of these files switched?

Reply:
@anirudh723 @varunthakkar1 You know this code a lot better than me and I dont want to mess anything up, do you know what code to replace here?